federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPContextManager.java

```diff
@@ -76,15 +80,22 @@ private void createLdapContext() throws NamingException {
 
         ldapContext = new InitialLdapContext(connProp, null);
         if (ldapConfig.isStartTls()) {
+            SSLSocketFactory sslSocketFactory = null;
+            String useTruststoreSpi = ldapConfig.getUseTruststoreSpi();
+            if (useTruststoreSpi != null && useTruststoreSpi.equals(LDAPConstants.USE_TRUSTSTORE_ALWAYS)) {
+                TruststoreProviderFactory truststoreFactory = (TruststoreProviderFactory) session.getKeycloakSessionFactory().getProviderFactory(TruststoreProvider.class, "file");
```

Comment:
I think that lines 86-87 should be replaced with:
````
TruststoreProvider provider = session.getProvider(TruststoreProvider.class);
````
This is more proper use of KeycloakSession SPI and at the same time, it is not hardcoded to "file" provider - the users can in theory replace the "file" provider with something else

Reply:
@mposolda Thanks for the review & suggestion of the fix! (didn't know there's more easier / appropriate way on how to achieve the same). Both instances fixed per your suggestion